### ISSUE TITLE
Problems with playlist sync

### ISSUE THREAD

**Issue by hetas** [lukedirtwalker/cutespotify#35]
At some point my playlists have stopped syncing from CuteSpot to other devices and other way around: changes made to playlists on other devices don't sync to CuteSpot.

I tried deleting all offline data and settings on the phone and on initial sync all the latest playlists were synced to the phone but subsequent changes were not.

At least not all of them. I tested with my devices and a few changes made with CuteSpot synced to all devices but most of them didn't. Same thing other way around. Changes made with Linux official Spotify-client synced to Android-spotify-client running on Jolla but not to CuteSpot.

Tested with Jolla 1, Sailfish OS 2.0.1.11 and CuteSpot 1.5.7-1 over WLAN and
Jolla C, Sailfish OS 2.0.2.43 and CuteSpot 1.5.7-1 over WLAN and 4G.

**Comment by hetas** [comment]
Here's some log. There's at least some of these channel errors if they have something to do with the problem.

[cutespot.log.txt](https://github.com/lukedirtwalker/cutespotify/files/542652/cutespot.log.txt)

`
...

18:39:55.615 I [offline-mgr:2082] Storage has been cleaned

18:39:55.703 E [ap:4210] ChannelError(8, 1, playlist)

18:39:55.744 E [ap:4210] ChannelError(10, 1, playlist)

18:39:55.745 E [ap:4210] ChannelError(11, 0, playlist)

18:39:55.745 E [ap:4210] ChannelError(15, 1, playlist)

18:39:55.745 E [ap:4210] ChannelError(14, 1, playlist)

18:39:55.745 E [ap:4210] ChannelError(13, 1, playlist)

18:39:55.800 E [ap:4210] ChannelError(12, 1, playlist)

18:39:55.801 E [ap:4210] ChannelError(5, 1, link-tracks)

[D] QSpotifySession::event:437 - Process spotify event 
[D] QSpotifySession::processSpotifyEvents:567 - QSpotifySession::processSpotifyEvents 
[D] QSpotifyCacheManager::cacheInfo:113 - #Cache Info: Tracks 0 Artists 0 Ablums 0 
[D] QSpotifySession::processSpotifyEvents:579 - Processing events... 
[D] callback_notify_main_thread:106 - Notify main thread 
[D] QSpotifySession::event:450 - Connection error 
[D] QSpotifySession::setConnectionError:672 - QSpotifySession::setConnectionError 0 "No error" 
[D] QSpotifySession::event:511 - Logged in 1 
[D] QSpotifySession::onLoggedIn:615 - Logged in 
18:39:56.057 E [ap:4210] ChannelError(5, 1, link-tracks)

18:40:04.117 E [ap:4210] ChannelError(9, 1, playlist)

...

[D] QSpotifyCacheManager::cacheInfo:113 - #Cache Info: Tracks 9232 Artists 5173 Ablums 7455 
[D] QSpotifySession::processSpotifyEvents:579 - Processing events... 
[D] callback_userinfo_updated:118 - User info updated 
[D] QSpotifySession::event:456 - Meta data 
18:40:18.551 E [ap:4210] ChannelError(0, 1, playlist)

[D] callback_notify_main_thread:106 - Notify main thread 
[D] QSpotifySession::event:456 - Meta data 
18:40:18.661 E [ap:4210] ChannelError(1, 1, link-tracks)

[D] QSpotifySession::event:456 - Meta data 
[D] QSpotifySession::event:437 - Process spotify event 
[D] QSpotifySession::processSpotifyEvents:567 - QSpotifySession::processSpotifyEvents 
[D] QSpotifyCacheManager::cacheInfo:113 - #Cache Info: Tracks 9232 Artists 5173 Ablums 7455 
[D] QSpotifySession::processSpotifyEvents:579 - Processing events... 
[D] callback_notify_main_thread:106 - Notify main thread 
[D] QSpotifySession::event:456 - Meta data 
[D] QSpotifySession::event:437 - Process spotify event 

...
`

**Comment by lukedirtwalker** [comment]
Hm as I said on talk.maemo.org I can only agree that there is currently a problem. But I can't do anything against it. Spotify has stopped supporting libspotify without providing a replacement currently. We can only wait until they release the new library and the a rewrite of custespot is required.

**Comment by hetas** [comment]
Ok, sorry, I didn't realise libspotify support has been dropped. That's a shame, I really like Cutespot. I've donated to this project and I'm going to donate more if there's future for this project. Unfortunately Spotify seems to be putting all their effort in Android an iOS instead of public APIs.

For this playlist problem [WepApis playlist management](https://developer.spotify.com/web-api/working-with-playlists/) might be an alternative. Search seems to be broken now also and that could be implemented using [WepApi](https://developer.spotify.com/technologies/metadata-api/search/).

If you, or someone else, is willing to go that way while there's now guarantees that there's going to be a replacement for libspotify.

Thanks for your effort.

EDIT: Search is working again. Problem might have have a been related to [ddos attack on dyn](https://threatpost.com/dyn-confirms-ddos-attack-affecting-twitter-github-many-others/121438/).

**Comment by lukedirtwalker** [comment]
Yeah WebApi is definitely the way forward. But I don't think it is worth integrating it with libspotify. I'd rather integrate it with the new libspotify embedded once it's available.

**Comment by bonanza123** [comment]
@lukedirtwalker is it official that they are working on a new lib?

**Comment by lukedirtwalker** [comment]
See https://developer.spotify.com/technologies/libspotify/ :

> We hope to be able to provide you with a new library for other platforms during 2016.

And if you look into the AndroidSDK you see a libspotify_embedded.so which I guess will be the one they'll release.

**Comment by BonoNL** [comment]
I will spam Spotify to make them work some faster!

**Comment by BonoNL** [comment]
Replay from Spotify:

Hey,
there won't be a publicly available libspotify successor before the end of 2016. Happy to add you to a list of potential users of a private beta, if we run one. Please email developer-support@spotify.com with what you'd like to use it for 🙂 Happy holidays!

**Comment by lukedirtwalker** [comment]
Hi @BonoNL  Thanks, I will write them.

**Comment by lukedirtwalker** [comment]
Spotify said they might invite me to a private Beta if there will be one. That was nearly 2 months ago, haven't heard back from them.

**Comment by BonoNL** [comment]
I will mail them and ask how the case is.
This is not the way to threat customers and developers.

EDIT:

I've mailed them with the question to hurry up.

**Comment by hetas** [comment]
Syncing seems to work better now. I don't if it has something to do with the Sailfish OS 2.1-version. Spotify-provided Discovery Weekly and Release Radar lists don't work though.

**Comment by bonanza123** [comment]
@BonoNL @lukedirtwalker any updates regarding the library from spotify?

**Comment by BonoNL** [comment]
I never heard from those guys at Spotify again, and how about you @lukedirtwalker?

**Comment by lukedirtwalker** [comment]
nope, unfortunately not.

**Comment by ahjolinna** [comment]
@lukedirtwalker : have you checked [Spotifyd](https://github.com/Spotifyd/spotifyd) project, not sure how feature wide it is.

**Comment by lukedirtwalker** [comment]
@ahjolinna hey no I haven't. It is written in Rust not sure how that would work on Sailfish, but other than that it seems like a cool project. Anyway I don't really have too much time to work on Cutespot anymore.
